### PR TITLE
Remove circular dependency between default otel and connector

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: connector/metrics_router.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package connector // import "go.opentelemetry.io/collector/connector"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/internal/fanoutconsumer"
```
Comment (dmitryax) [inline connector/metrics_router.go:9]: the linter should be failing here

File: connector/logs_router_test.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+				assert.Len(t, logs, expected[id])
+				for n := 0; n < len(logs); n++ {
+					assert.EqualValues(t, ld, logs[n])
+				}
+			}
+		}
+	}
+}
+
+func TestLogsGetConsumers(t *testing.T) {
```
Comment (dmitryax) [inline connector/logs_router_test.go:107]: nit: the test names are inconsistent between between data types. Should be `Test[Logs|Metrics|Traces]RouterConsumer`?

**Review by dmitryax — APPROVED** [review]

File: connector/metrics_router.go (hunk 1 above)
Comment (bogdandrutu) [reply connector/metrics_router.go]: Done.

File: connector/logs_router_test.go (hunk 2 above)
Comment (bogdandrutu) [reply connector/logs_router_test.go]: Done.

**Review by dmitryax — APPROVED** [review]

**Review by djaglowski — COMMENTED** [review]
I'm working on something closely related and would like some time to reconcile possible changes with this PR. I may have some suggestions based on this perspective. I will follow up here today or tomorrow.

**Review by djaglowski — APPROVED** [review]
This looks good to me. The changes I am looking at elsewhere would fit into this just as well either way.

**Review by codeboten — COMMENTED** [review]
I believe this change should be done by first deprecating the original interfaces no? https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#breaking-changes

**Comment by djaglowski** [comment]
Hmm, the intention isn't to remove the interfaces, but changing them does break them. Do we deprecate, then remove the deprecation while making the change?

**Comment by codeboten** [comment]
> Hmm, the intention isn't to remove the interfaces, but changing them does break them. Do we deprecate, then remove the deprecation while making the change?

I believe this was done in the past by providing a temporary interface name, that will then be renamed again

**Comment by bogdandrutu** [comment]
@codeboten I think this has a very limited use, and the benefit of doing the deprecation is limited. I would actually ask to go with an exception if that is the case.

**Pull request merged**